exporter/keys.go

```diff
@@ -235,7 +202,55 @@ func parseKeyArg(keysArgString string) (keys []dbKeyPair, err error) {
 			return keys, fmt.Errorf("couldn't parse db/key string: %s", k)
 		}
 
+		// We want to guarantee at the top level that invalid values
+		// will not fall into the final Redis call.
+		if db == "" && key == "" {
+			log.Debugf("parseKeyArg(): Got empty database and key, pair skipped")
+			continue
+		} else if db == "" || key == "" {
+			log.Errorf("parseKeyArg(): Empty value parsed in pair '%s=%s', skip", db, key)
+			continue
+		}
+
+		number, err := strconv.Atoi(db)
+		if err != nil {
```

Comment:
Similar here, just check for `err != nil || number < 0` and return `"invalid db index", that's good enough and simplifies things a little.